include/dlaf/factorization/qr/t_factor_impl.h

```diff
@@ -146,26 +143,28 @@ struct Helpers<Backend::GPU, Device::GPU, T> {
 
   template <class VISender, class TSender>
   static auto gemvColumnT(SizeType first_row_tile, VISender&& tile_vi,
-                          pika::shared_future<common::internal::vector<T>>& taus,
-                          TSender&& tile_t) noexcept {
+                          matrix::ReadOnlyTileSender<T, Device::CPU> taus, TSender&& tile_t) noexcept {
     namespace ex = pika::execution::experimental;
 
     auto gemv_func = [first_row_tile](cublasHandle_t handle, const auto& tile_v, const auto& taus,
                                       auto& tile_t) noexcept {
       const SizeType k = tile_t.size().cols();
       DLAF_ASSERT(tile_v.size().cols() == k, tile_v.size().cols(), k);
-      DLAF_ASSERT(taus.size() == k, taus.size(), k);
+      DLAF_ASSERT(taus.size().rows() == k, taus.size().rows(), k);
 
       if (first_row_tile == 0) {
         whip::stream_t stream;
         DLAF_GPUBLAS_CHECK_ERROR(cublasGetStream(handle, &stream));
 
-        whip::memcpy_2d_async(tile_t.ptr(), to_sizet(tile_t.ld() + 1) * sizeof(T), taus.data(),
-                              sizeof(T), sizeof(T), to_sizet(k), whip::memcpy_default, stream);
+        // Assume taus is a column vector with column major layout, i.e. elements are contiguous
+        // TODO: Can this be asserted more directly?
```

Comment:
Am I missing some handy way to assert that a tile is column-major and has only one column? As far as I can tell we implicitly assume that all matrices are column major.

Reply:
Tiles are column major. In this case you only need to check that the size is (k x 1).

Reply:
Updated in https://github.com/eth-cscs/DLA-Future/pull/908/commits/61e43b0cb08a04ac7be968a6f3653c3aa0be6881. I left the comment and moved the assertion up to the top of the function along with the other assertions.